tests/test_reconciliation/test_base.py

```diff
@@ -105,3 +120,26 @@ def test_reconcile(hierarchical_ts, source_level, target_level, mapping_matrix,
     assert obtained_ts.regressors == expected_ts.regressors
     pd.testing.assert_frame_equal(obtained_ts.df, expected_ts.df)
     pd.testing.assert_frame_equal(obtained_ts.df_exog, expected_ts.df_exog)
+
+
+@pytest.mark.parametrize(
+    "source_level, target_level, mapping_matrix",
+    [("market", "total", "market_total_mapping_matrix"), ("total", "market", "total_market_mapping_matrix")],
+)
+def test_reconcile_with_target_components(
+    hierarchical_ts_with_target_components, source_level, target_level, mapping_matrix, request
+):
+    source_ts = hierarchical_ts_with_target_components.get_level_dataset(target_level=source_level)
+    expected_ts = hierarchical_ts_with_target_components.get_level_dataset(target_level=target_level)
+
+    reconciliator = DummyReconciliator(target_level=target_level, source_level=source_level)
+    reconciliator.mapping_matrix = request.getfixturevalue(mapping_matrix)
+    obtained_ts = reconciliator.reconcile(ts=source_ts)
+
+    assert obtained_ts.freq == expected_ts.freq
```

Comment:
We should probably define `__eq__` in our `TSDataset`...

Reply:
Tried to do so earlier but there is a problem with comparing dtaframes